Source/WTF/wtf/DateMath.cpp

```diff
@@ -836,7 +836,12 @@ double parseDateFromNullTerminatedCharacters(const char* dateString, bool& isLoc
             year = std::nullopt;
         } else {
             // in the normal case (we parsed the year), advance to the next number
-            dateString = ++newPosStr;
+            // ' at 23:12:40 GMT'
+            if (isASCIISpace(*newPosStr) && toASCIILower(*(newPosStr + 1)) == 'a' && toASCIILower(*(newPosStr + 2)) == 't')
```

Comment:
`ASCIICType.h` has a faster idiom for such things with one fewer branch, although the function name is kind of long. Also, f it was me I would use I would use array-style indexing:

`if (isASCIISpace(newPosStr[0]) && isASCIIAlphaCaselessEqual(newPosStr[1], 'a') && isASCIIAlphaCaselessEqual(newPosStr[2], 't'))`

Reply:
Nice, fixed!